lib/collection/src/shards/local_shard/mod.rs

```diff
@@ -934,85 +932,105 @@ impl Drop for LocalShard {
 /// Convenience struct for combining clock maps belonging to a shard
 ///
 /// Holds a clock map for tracking the highest clocks and the cutoff clocks.
-#[derive(Clone)]
+#[derive(Clone, Debug, Default)]
 pub struct LocalShardClocks {
-    highest_clocks: Arc<Mutex<ClockMap>>,
-    cutoff_clocks: Arc<Mutex<ClockMap>>,
-    highest_clocks_path: PathBuf,
-    cutoff_clocks_path: PathBuf,
+    newest_observed_clocks: Arc<Mutex<ClockMap>>,
+    oldest_resolvable_clocks: Arc<Mutex<ClockMap>>,
 }
 
 impl LocalShardClocks {
-    pub fn new_at(highest_clocks: ClockMap, cutoff_clocks: ClockMap, shard_path: &Path) -> Self {
+    fn new(newest_observed_clocks: ClockMap, oldest_resolvable_clocks: ClockMap) -> Self {
         Self {
-            highest_clocks: Arc::new(Mutex::new(highest_clocks)),
-            cutoff_clocks: Arc::new(Mutex::new(cutoff_clocks)),
-            highest_clocks_path: Self::highest_clock_map_path(shard_path),
-            cutoff_clocks_path: Self::cutoff_clock_map_path(shard_path),
+            newest_observed_clocks: Arc::new(Mutex::new(newest_observed_clocks)),
+            oldest_resolvable_clocks: Arc::new(Mutex::new(oldest_resolvable_clocks)),
         }
     }
 
+    // Load clock maps from disk
+    pub fn load(shard_path: &Path) -> CollectionResult<Self> {
+        let newest_observed_clocks =
+            ClockMap::load_or_default(&Self::newest_observed_clock_map_path(shard_path))?;
+
+        let oldest_resolvable_clocks =
+            ClockMap::load_or_default(&Self::oldest_resolvable_clock_map_path(shard_path))?;
+
+        Ok(Self::new(newest_observed_clocks, oldest_resolvable_clocks))
+    }
+
     /// Persist clock maps to disk
-    pub async fn store(&self) -> CollectionResult<()> {
-        self.cutoff_clocks
+    pub async fn store(&self, shard_path: &Path) -> CollectionResult<()> {
+        // TODO: Is order important here!? If it is, it should be documented with a comment!
```

Comment:
My reasoning was that the oldest recoverable are more important than the newest seen with regards to consistency.

But I'm not sure if that's actually true.

Before I locked both together but that would be a potential foot gun for a deadlock. I left a comment regarding this, but it's gone for some reason.

Reply:
I think we can remove this TODO.